lib/resty/websocket/client.lua

```diff
@@ -339,6 +342,43 @@ function _M.connect(self, uri, opts)
         return nil, "unexpected HTTP response code: " .. m[1], header
     end
 
+    -- gather all response headers
+
+    local iter, err = re_gmatch(header .. "\r\n", "([^:\\s]+):\\s*(.*?)\r\n", "jo")
```

Comment:
I think we should not expect a header name to have any leading and trailing LWS (linear white space). If it has, this does not follow the RFC.

https://datatracker.ietf.org/doc/html/rfc2616#page-31:~:text=the%20common%20forms.-,message,-%2Dheader%20%3D%20field%2Dname

So we should get spaces from regex, but check if there any space in the header_name.

Reply:
This regex is copied from lua-resty-http, I think that it has already considered this issue.

Reply:
``` lua
local LINES = "   h0: v0\r\nh1: v1" .. "\r\n"

local iter = ngx.re.gmatch(LINES, "([^:\\s]+):\\s*(.*?)\r\n", "jo")

for m, err in iter do
    if err then
        ngx.say("error: ", err)
        return
    end

    ngx.say("key: ", m[1], ", value: ", m[2])
end
```

``` bash
$ resty test.lua
key: h0, value: v0
key: h1, value: v1
```

It is not correct.

Reply:
After offline discussion, since this API is only used in our MTLS connections, which should not have an unexpected request format, I'm ok for now.